llvm/utils/TableGen/DecoderEmitter.cpp

```diff
@@ -1438,19 +1365,11 @@ void FilterChooser::emitSoftFailTableEntry(DecoderTableInfo &TableInfo,
   APInt PositiveMask(BitWidth, 0ULL);
   APInt NegativeMask(BitWidth, 0ULL);
   for (unsigned i = 0; i < BitWidth; ++i) {
-    BitValue B(*SFBits, i);
-    BitValue IB(*InstBits, i);
-
-    if (B != BitValue::BIT_TRUE)
+    if (!isa<BitInit>(SFBits->getBit(i)) ||
+        !cast<BitInit>(SFBits->getBit(i))->getValue())
```

Comment:
Double casting isn't ideal, but this is going to be removed in the next patch anyway.